packages/store/src/patterns/interface-tools.js

```diff
@@ -144,6 +144,7 @@ const bindMethod = (
   let { method } = thisfulMethods
     ? {
         method(...args) {
+          this || Fail`thisful method ${methodTag} called without this`;
```

Comment:
```suggestion
          this === undefined || Fail`thisful method ${methodTag} called without this`;
```
Equivalent in practice. But it is possible to call the method with `this` explicitly bound to some other falsy value, for which this diagnostic is no longer correct.

In any case, I understand now why `getContext` cannot distinguish this case, and so cannot give as good a diagnostic. Thanks.

Reply:
I'd like to error in those cases too so I've changed the error message instead.